device.go

```diff
@@ -90,10 +103,35 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 		return syscall.ENOSPC
 	}
 
-	w, err := io.Copy(b, image)
+	ssz, err := b.SectorSize()
 	if err != nil {
-		log.Errorf("failed to write image data to device %v: %v",
+		log.Errorf("failed to read sector size of block device %s: %v",
 			inactivePartition, err)
+		return err
+	}
+
+	buf := make([]byte, ssz)
```

Comment:
Maybe more effective would be using `io.CopyN()` instead. Then we can simplify reading and writing to something like: 
```
	for {
		_, err := io.CopyN(b, image, ssz)

		if err == io.EOF {
			break
		} else if err != nil {
			log.Errorf("failed to read image data: %v", err)
			return err
		}
	}
```